docs/main/eventing-communication/eventing-automate-install.sh

```diff
@@ -0,0 +1,128 @@
+    echo -e "Usage: ORCHESTRATOR_NAME=ORCHESTRATOR_NAME BROKER_NAME=BROKER_NAME BROKER_NAMESPACE=BROKER_NAMESPACE [KAFKA_REPLICATION_FACTOR=KAFKA_REPLICATION_FACTOR] [ORCHESTRATOR_NAMESPACE=openshift-operators] [BROKER_TYPE=kafka] [INSTALL_KAFKA_CLUSTER=true] $program_name"
+    echo "  ORCHESTRATOR_NAME                   Name of the installed orchestrator CR"
+    echo "  ORCHESTRATOR_NAMESPACE              Optional, namespace in which the orchestrator operator is depoyed. Default is openshift-operators"
+    echo "  BROKER_NAME                         Name of the broker to install"
+    echo "  BROKER_NAMESPACE                    Namespace in which the broker must be installed"
+    echo "  BROKER_TYPE                         Optional , type of the broker. Either 'kafka' or 'in-memory'. Default is: 'kafka'"
+    echo "  INSTALL_KAFKA_CLUSTER               Optional, if set to true, indicates that kafka cluster must be installed. Will only be used if BROKER_TYPE is 'kafka'. Default is: true"
+    echo "  KAFKA_REPLICATION_FACTOR            Optional, only used if INSTALL_KAFKA_CLUSTER is set to false and BROKER_TYPE is 'kafka', provide the replication factor for the Kafka cluster"
+    exit 1
+}
+
+if [[ -z "${ORCHESTRATOR_NAMESPACE}" ]]; then
+  ORCHESTRATOR_NAMESPACE=openshift-operators
+fi
+
+if [[ -z "${ORCHESTRATOR_NAME}" ]]; then
+  echo "ORCHESTRATOR_NAME env variable must be set to the name of the deployed orchestrator CR; e.g: orchestrator-sample"
+  usage
+fi
+
+if [[ -z "${BROKER_TYPE}" ]]; then
+  BROKER_TYPE=kafka
+fi
+
+if [[ "$BROKER_TYPE" != "kafka" && "$BROKER_TYPE" != "in-memory" ]]; then
+  echo 'Error: BROKER_TYPE env variable must be set to either "kafka" or "in-memory"'
+  usage
+fi
+
+if [[ -z "${INSTALL_KAFKA_CLUSTER}" ]]; then
+  if [ "$BROKER_TYPE" == "kafka" ]; then
+    INSTALL_KAFKA_CLUSTER=true
+  else
+    INSTALL_KAFKA_CLUSTER=false
+  fi
+fi
+
+if [[ -z "${BROKER_NAME}" ]]; then
+  echo "BROKER_NAME env variable must be set to the name of the broker; e.g: kafka-broker."
+  usage
+fi
+
+if [[ -z "${BROKER_NAMESPACE}" ]]; then
+  echo "BROKER_NAMESPACE env variable must be set to the namespace of the broker; e.g: ${TARGET_NS}."
+  usage
+fi
+
+if [ "$INSTALL_KAFKA_CLUSTER" == true ]; then
+    echo "Installing a kafka cluster using Strimzi default quickstart: https://strimzi.io/quickstarts/"
+    oc create namespace kafka
+    oc create -f 'https://strimzi.io/install/latest?namespace=kafka' -n kafka
+    oc apply -f https://strimzi.io/examples/latest/kafka/kraft/kafka-single-node.yaml -n kafka 
+    oc wait kafka/my-cluster --for=condition=Ready --timeout=300s -n kafka 
+    KAFKA_REPLICATION_FACTOR=1
+fi
+
+if [ "$BROKER_TYPE" == "kafka" ]; then
+    if [[ -z "${KAFKA_REPLICATION_FACTOR}" ]]; then
+      echo "KAFKA_REPLICATION_FACTOR env variable must be set when INSTALL_KAFKA_CLUSTER is set to false and BROKER_TYPE is 'kafka'."
+      usage
+    fi
+    echo "Installing Knative 1.16.1 kafka-related resources"
```

Comment:
perhaps parameterize the version to ease maintenance of the script ?

Reply:
1.17.0 will not work on older OCP version; my demo cluster is on 4.16 and when I use 1.17.0 I get an error stating that only k8s version >= 1.30 are supported

But I will create a variable with the version in case we want to test to upgrade more easily